sdk/src/androidTest/java/com/mapbox/maps/ViewAnnotationTest.kt

```diff
@@ -0,0 +1,683 @@
+          allowOverlap(false)
+        }
+      )
+      val secondView = viewAnnotationManager.addViewAnnotation(
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(SHIFTED_CENTER)
+          allowOverlap(false)
+        }
+      )
+      mainHandler.postDelayed(
+        {
+          assertFalse(mapView.hasChildView(firstView))
+          assertTrue(mapView.hasChildView(secondView))
+          latch.countDown()
+          viewAnnotationManager.removeViewAnnotation(secondView)
+          mainHandler.postDelayed(
+            {
+              assertTrue(mapView.hasChildView(firstView))
+              assertFalse(mapView.hasChildView(secondView))
+              latch.countDown()
+            },
+            VIEW_PLACEMENT_DELAY_MS
+          )
+        },
+        VIEW_PLACEMENT_DELAY_MS
+      )
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  // checking some use-cases when using associatedFeatureId
+
+  @Test
+  fun associatedFeatureIdWhenFeatureVisible() {
+    val latch = CountDownLatch(1)
+    mainHandler.post {
+      mapboxMap.getStyle {
+        prepareStyle(it, Visibility.VISIBLE)
+        val view = viewAnnotationManager.addViewAnnotation(
+          resId = R.layout.view_annotation,
+          options = viewAnnotationOptions {
+            geometry(CAMERA_CENTER)
+            anchor(ViewAnnotationAnchor.BOTTOM)
+            offsetY(50)
+            associatedFeatureId(ASSOCIATED_FEATURE_ID)
+          }
+        )
+        mainHandler.postDelayed(
+          {
+            assertTrue(mapView.hasChildView(view))
+            latch.countDown()
+          },
+          VIEW_PLACEMENT_DELAY_MS
+        )
+      }
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun associatedFeatureIdWhenFeatureGone() {
+    val latch = CountDownLatch(1)
+    mainHandler.post {
+      mapboxMap.getStyle {
+        prepareStyle(it, Visibility.NONE)
+        val view = viewAnnotationManager.addViewAnnotation(
+          resId = R.layout.view_annotation,
+          options = viewAnnotationOptions {
+            geometry(CAMERA_CENTER)
+            anchor(ViewAnnotationAnchor.BOTTOM)
+            offsetY(50)
+            associatedFeatureId(ASSOCIATED_FEATURE_ID)
+          }
+        )
+        mainHandler.postDelayed(
+          {
+            assertFalse(mapView.hasChildView(view))
+            latch.countDown()
+          },
+          VIEW_PLACEMENT_DELAY_MS
+        )
+      }
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  private fun MapView.hasChildView(view: View): Boolean {
+    children.forEach { v ->
+      if (v == view) {
+        return true
+      }
+    }
+    return false
+  }
+
+  private fun MapView.getChildViewIndex(view: View): Int {
```

Comment:
nit:
maybe move to some utils class/file?

Reply:
done